conformance/lang/expressions/logical-expr/logical_and.balt

```diff
@@ -0,0 +1,699 @@
+    io:println(4 < 2 && true); // @output false
+    io:println(false && 4 < 2); // @output false
+    io:println(4 < 2 && false); // @output false
+    io:println(4 < 3 && 4 < 3); // @output false
+
+    io:println(true && 4 >= 2); // @output true
+    io:println(4 >= 2 && true); // @output true
+    io:println(false && 4 >= 2); // @output false
+    io:println(4 >= 2 && false); // @output false
+    io:println(4 >= 3 && 4 >= 3); // @output true
+
+    io:println(true && 4 <= 2); // @output false
+    io:println(4 <= 2 && true); // @output false
+    io:println(false && 4 <= 2); // @output false
+    io:println(4 <= 2 && false); // @output false
+    io:println(4 <= 3 && 4 <= 3); // @output false
+
+    io:println(4 < 2 && 4 > 2); // @output false
+    io:println(4 >= 2 && 4 <= 3); // @output false
+    io:println(4 > 2 && 4 <= 2); // @output false
+    io:println(4 >= 2 && 4 < 2); // @output false
+}
+
+Test-Case: output
+Description: Test logical expression using is expression.
+Labels: logical-expr, is-expr
+
+function init() {
+    io:println(true && 2 is int); // @output true
+    io:println(2 is int && true); // @output true
+    io:println(false && 2 is int); // @output false
+    io:println(2 is int && false); // @output false
+    io:println(2 is int && 2 is int); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using equality expression.
+Labels: logical-expr, equality
+
+function init() {
+    io:println(true && 2 == 3); // @output false
+    io:println(3 == 3 && true); // @output true
+    io:println(false && 3 == 3); // @output false
+    io:println(2 == 3 && false); // @output false
+    io:println(2 == 3 && 3 == 3); // @output false
+
+    io:println(true && 2 != 3); // @output true
+    io:println(3 != 3 && true); // @output false
+    io:println(false && 3 != 3); // @output false
+    io:println(2 != 3 && false); // @output false
+    io:println(2 != 3 && 3 != 3); // @output false
+
+    io:println(true && 2 === 3); // @output false
+    io:println(3 === 3 && true); // @output true
+    io:println(false && 3 === 3); // @output false
+    io:println(2 === 3 && false); // @output false
+    io:println(2 === 3 && 3 === 3); // @output false
+
+    io:println(true && 2 !== 3); // @output true
+    io:println(3 !== 3 && true); // @output false
+    io:println(false && 3 !== 3); // @output false
+    io:println(2 !== 3 && false); // @output false
+    io:println(2 !== 3 && 3 !== 3); // @output false
+}
+
+Test-Case: output
+Description: Test logical expression with logical expression.
+Labels: logical-expr
+
+function init() {
+    io:println(true && true && false); // @output false
+    io:println(true && false && true); // @output false
+    io:println(false && true && false); // @output false
+    io:println(true && false && false); // @output false
+    io:println(true && false && true && false); // @output false
+}
+
+Test-Case: output
+Description: Test evaluation order of logical expression.
```

Comment:
Let's elaborate a bit more on what exactly is expected here. We shouldn't have to check the spec to identify that IMO.